src/lib/transaction_logic/test/transaction_test_lib.ml

```diff
@@ -0,0 +1,119 @@
+open Base
+open Mina_transaction_logic
+
+module type Testable_boolean_logic = sig
+  include Bool_intf
+
+  val eval : t -> bool
+end
+
+(* This functor provides basic tests for modules of the signature
+   Zkapp_command_logic.Bool_intf passed as inputs to the
+   Zkapp_command_logic.Make functor. It checks all the basic
+   properties of boolean algebra.
+
+   Users of Zkapp_command_logic.Make outside this package need to
+   construct this functor themselves in order to test their input
+   modules. *)
+module Logic_test (Logic : Testable_boolean_logic) = struct
+  module Truth_table_test : sig
+    include Monad.S
+
+    val var : int -> Logic.t t
+
+    val exec : var_count:int -> ?expect:bool -> Logic.t t -> unit
+  end = struct
+    type 'a t = Logic.t array -> 'a
+
+    let make i =
+      let rec mk_lists acc = function
+        | i when i <= 0 ->
+            acc
+        | i ->
+            mk_lists
+              (List.concat_map acc ~f:(fun bs ->
+                   Logic.[ true_ :: bs; false_ :: bs ] ) )
+              (i - 1)
+      in
+      List.map ~f:Array.of_list @@ mk_lists [ [] ] i
+
+    include Monad.Make (struct
+      type nonrec 'a t = 'a t
+
+      let map = `Custom (fun t ~f bs -> f (t bs))
+
+      let return = Fn.const
+
+      let bind t ~f bs = f (t bs) bs
+    end)
+
+    let var i bs = Array.get bs i
+
+    let exec ~(var_count : int) ?(expect = true) (test : Logic.t t) =
+      List.iter (make var_count) ~f:(fun bs ->
+          [%test_result: bool] (Logic.eval @@ test bs) ~expect )
+  end
+
+  let%test_module "Logic tests" =
```

Comment:
Not sure these tests are needed here. Surely these properties hold for the OCaml `bool` type. 

For the Snarky type `Boolean.var`, used in `Transaction_snark.Inputs`, wouldn't it be better to have tests in Snarky. The tests seem unrelated to transaction logic as such.

Reply:
You're probably right. I already tried to instantiate the module for `Boolean.var`, but it's not straightforward, as there's no evaluation function I could find in the scope. I supposse I'd have to import more packages from `snarky`, in which case it's likely better to put these tests directly in `snarky`. I'll discard this commit and write appropriate tests in `snarky`.

Reply:
I removed these tests. I'll add those for `snarky` in a separate PR.

Reply:
I would also suggest that only the truth tables for ||, &&, and NOT would need testing; properties like associativity, De Morgan laws, etc., follow directly from those.